assets/src/scripts/components/hydrograph/time-series-graph.spec.js

```diff
@@ -0,0 +1,388 @@
+                timeSeriesCollections: ['coll2', 'col4']
+            }
+        },
+        variables: {
+            '45807197': {
+                variableCode: {
+                    value: '00060'
+                },
+                oid: '45807197',
+                variableName: 'Test title for 00060',
+                variableDescription: 'Test description for 00060',
+                unit: {
+                    unitCode: 'unitCode'
+                }
+            },
+            '45807190': {
+                variableCode: {
+                    value: '00010'
+                },
+                oid: '45807190',
+                unit: {
+                    unitCode: 'unitCode'
+                }
+            }
+        },
+        methods: {
+            'method1': {
+                methodDescription: 'method description'
+            }
+        }
+    },
+    statisticsData : {
+        median: {
+            '00060': {
+                '1234': [
+                    {
+                        month_nu: '2',
+                        day_nu: '20',
+                        p50_va: '40',
+                        begin_yr: '1970',
+                        end_yr: '2017',
+                        loc_web_ds: 'This method'
+                    }, {
+                        month_nu: '2',
+                        day_nu: '21',
+                        p50_va: '41',
+                        begin_yr: '1970',
+                        end_yr: '2017',
+                        loc_web_ds: 'This method'
+                    }, {
+                        month_nu: '2',
+                        day_nu: '22',
+                        p50_va: '42',
+                        begin_yr: '1970',
+                        end_yr: '2017',
+                        loc_web_ds: 'This method'
+                    }
+                ]
+            }
+        }
+    },
+    timeSeriesState: {
+        currentVariableID: '45807197',
+        currentDateRange: 'P7D',
+        requestedTimeRange: null,
+        showSeries: {
+            current: true,
+            compare: true,
+            median: true
+        },
+        loadingTSKeys: []
+    },
+    ui: {
+        width: 400
+    }
+};
+
+describe('time series graph', () => {
+    let graphNode;
+
+    beforeEach(() => {
+        let body = select('body');
+        let component = body.append('div')
+            .attr('id', 'hydrograph');
+        component.append('div').attr('class', 'loading-indicator-container');
+        component.append('div').attr('class', 'graph-container');
+        component.append('div').attr('class', 'select-time-series-container');
+        component.append('div').attr('class', 'provisional-data-alert');
+
+        graphNode = document.getElementById('hydrograph');
+
+        jasmine.Ajax.install();
+    });
+
+    afterEach(() => {
+        jasmine.Ajax.uninstall();
+        select('#hydrograph').remove();
+    });
+
+    it('empty graph displays warning', () => {
```

Comment:
Lines 175 to... 188 or possibly a few of the early tests might be considered initialization tests and could maybe stay in the index.spec.

Reply:
I agree that they belong there. I think it will be clearer what belongs here once you call the exporting function in time-series-graph.js